### PR TITLE
[BEAM-927] Fix findbugs and re-enable Maven plugin in JmsIO

### PR DESCRIPTION
**Pull request by jbonofre**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [X] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @dhalperi

File: sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsRecord.java [hunk 1]
```diff
@@ -31,8 +30,8 @@
   private final String jmsMessageID;
   private final long jmsTimestamp;
   private final String jmsCorrelationID;
-  private final Destination jmsReplyTo;
-  private final Destination jmsDestination;
+  private final String jmsReplyTo;
+  private final String jmsDestination;
```
Comment (dhalperi) [inline sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsRecord.java]: Per the [documentation](http://docs.oracle.com/javaee/6/api/javax/jms/Destination.html):

> JMS provider implementations of administered objects should implement the `javax.naming.Referenceable` and `java.io.Serializable` interfaces so that they can be stored in all JNDI naming contexts.

Up to you, but it might be okay to leave the type as `Destination` and instead suppress the Findbugs message, documenting the reason. E.g., https://github.com/apache/incubator-beam/blob/5864a38ba595cdf90d2d5559e4312ef6144f60c7/sdks/java/core/src/main/java/org/apache/beam/sdk/util/ExposedByteArrayInputStream.java#L36

Comment (dhalperi) [inline sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsRecord.java]: Mark nullable fields `@Nullable` here and in constructor and on getters.

Comment (jbonofre) [inline sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsRecord.java]: Yes, good idea. Let me update the PR.

Comment (jbonofre) [inline sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsRecord.java]: `Destination` can't be nullable, but `ReplyTo` can. I will update accordingly. Thanks !

**Comment by jbonofre** [comment]
Updated according to the last messages.

**Comment by jbonofre** [comment]
@dhalperi gently reminder ;)

**Comment by dhalperi** [comment]
Unfortunately (and sorry that I missed this at ApacheCon / then vacation) we have had to remove the Findbugs annotation jar from the SDK because it does not have an apache compatible license. We now have to use the findbugs suppression file to implement the filtering. :/

**Comment by jbonofre** [comment]
I'm updating accordingly, thanks !

**Comment by jbonofre** [comment]
Rebased and updated. I'm using `<onlyAnalyze/>` configuration to exclude `JmsRecord` from findbugs.

**Comment by jbonofre** [comment]
@dhalperi Gently reminder ;)

CC: @davorbonaci

File: sdks/java/io/jms/pom.xml [hunk 2]
```diff
@@ -59,6 +46,15 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-jar-plugin</artifactId>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>findbugs-maven-plugin</artifactId>
```
Comment (davorbonaci) [inline sdks/java/io/jms/pom.xml]: I think this is fine too, but I'd prefer that we have one way of suppressing false positives. Right now, all suppressions seem to be in `findbugs-filter.xml`, whereas this would introduce another way of suppressing some warnings.

I'd recommend to move the suppressions there, if possible.

**Comment by davorbonaci** [comment]
Also:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project beam-sdks-java-io-jms: Compilation failure: Compilation failure:
[ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_MavenInstall/sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsRecord.java:[20,39] package edu.umd.cs.findbugs.annotations does not exist
[ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_MavenInstall/sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsRecord.java:[33,2] cannot find symbol
[ERROR] symbol: class SuppressFBWarnings
```

File: sdks/java/io/jms/pom.xml (hunk 2 above)
Comment (jbonofre) [inline sdks/java/io/jms/pom.xml]: OK, as said in previous comment, I used `<onlyAnalyze/>` to avoid to create a new file. Instead I will use `findbugs-filter.xml`.

Comment (davorbonaci) [inline sdks/java/io/jms/pom.xml]: I don't think you need a new file. There's a `findbugs-filter.xml` already, which is automatically applied to all modules. Am I incorrect?

Comment (jbonofre) [inline sdks/java/io/jms/pom.xml]: Oh, you mean on the "main" `findbugs-filter.xml`. I thought about one specific in the `JmsIO`. OK, I update this way.

**Comment by jbonofre** [comment]
Rebased and updated according to Davor's comment.

File: sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsRecord.java [hunk 3]
```diff
@@ -31,7 +33,7 @@
   private final String jmsMessageID;
   private final long jmsTimestamp;
   private final String jmsCorrelationID;
-  private final Destination jmsReplyTo;
+  @Nullable private final Destination jmsReplyTo;
```
Comment (dhalperi) [inline sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsRecord.java]: propagate `@Nullable` to other uses? E.g., the constructor and the getter?

File: sdks/java/io/jms/pom.xml (hunk 2 above)
Comment (davorbonaci) [inline sdks/java/io/jms/pom.xml]: Is this addition needed now? I'm guessing that it isn't, since it should be inherited from the parent.

**Comment by jbonofre** [comment]
Yup, applying the proposed changes.

**Comment by jbonofre** [comment]
Updated according to the last comments.

**Comment by davorbonaci** [comment]
LGTM. Merging (the failure is known; other tests have been run).
